src/server/actions.js

```diff
@@ -15,20 +17,50 @@ class Actions {
     this.account = keyring.addFromMnemonic(mnemonic);
   }
 
-  async drip(address, amount) {
+  async processDrip ({ dripType, ...params})  {
+    switch (dripType) {
+      case DRIP_TYPE.NORMAL: 
+        await this.drip(params);
+      case DRIP_TYPE.LATER: 
+        return await this.dripLater(params);
+      case DRIP_TYPE.SWAG: 
+        return this.dripSwag(params);
+      default:
+    }
+    return null;
+  }
+
+  async drip({ address }) {
+    const amount = dripActions[DRIP_TYPE.NORMAL].amount * units;
     const transfer = this.api.tx.balances.transfer(address, amount);
     const hash = await transfer.signAndSend(this.account, { nonce: -1 });
     return { hash: hash.toHex() };
   }
 
-  async dripLater(address, amount, timestamp) {
+  async dripLater({ address, timestamp }) {
+    const { amount } = dripActions[DRIP_TYPE.LATER].amount * units;
     const providerId = uuid.v4();
 
     const extrinsic = this.api.tx.automationTime.scheduleNativeTransferTask(uuid.v4(), Math.floor(timestamp/1000), address, amount);
     const hash = await extrinsic.signAndSend(this.account, { nonce: -1 });
     return { hash: hash.toHex(), providerId };
   }
 
+  async dripSwag({ address }) {
+    const { amount } = dripActions[DRIP_TYPE.SWAG].amount * units;
+    const providerId = uuid.v4();
+
+    const now = moment().utc();
+    const executionTimes = [];
+    for (let i = 1; i <= 24; i += 1) {
+      executionTimes.push(Math.ceil(now.add(i, 'hours').valueOf() / 1000));
+    }
+
+    const extrinsic = this.api.tx.automationTime.scheduleNativeTransferTask(providerId, executionTimes, address, amount);
+    const hash = await extrinsic.signAndSend(this.account, { nonce: -1 });
```

Comment:
@justinzhou93 - this the appropriate way to manage the nonce?

Reply:
hm idk what a negative nonce means. nonce doesn't really matter unless you're making more than 1 call for a wallet while a call with an existing nonce is still in the transaction pool. so in this case, drip seems to be scheduling 1 task at a time for a given wallet, so therefore, no issue.

Reply:
https://polkadot.js.org/docs/api/cookbook/tx#how-do-i-take-the-pending-tx-pool-into-account-in-my-nonce

About "nonce = -1", this is an official example.

_As a convenience function, the accountNextIndex can be omitted by specifying a nonce of -1, allow the API to do the lookup. In this case the above can be simplified even further,_